### ISSUE TITLE
Generate element name with hyphen in it

### ISSUE THREAD

**Issue by wezm** [utkarshkukreti/markup.rs#19]
In order to use https://github.com/github/markdown-toolbar-element I need to generate markup that looks like this:

```html
<markdown-toolbar for="textarea_id">
  <md-bold>bold</md-bold>
  <md-header>header</md-header>
  <md-italic>italic</md-italic>
</markdown-toolbar>
```

I tried:

```rust
markdown-toolbar[for="body"] {
}
```

and

```rust
"markdown-toolbar"[for="body"] {
}
```

but neither worked. Is this possible with markup.rs? ~~If not, do you think the quoted syntax makes sense? If it does I could have a go at implementing it.~~ Edit: I realised the quoted string is a string literal already, so that won't work.

**Comment by wezm** [comment]
I have found a workaround for my project. Since I don't have a need for HTML elements with underscores in them I replace `_` with `-` when generating mark-up:

```diff
diff --git a/markup-proc-macro/src/generate.rs b/markup-proc-macro/src/generate.rs
index ccd6fbc..65d05c5 100644
--- a/markup-proc-macro/src/generate.rs
+++ b/markup-proc-macro/src/generate.rs
@@ -112,8 +112,9 @@ impl Generate for Element {
             children,
             close,
         } = self;
+        let element_name = name.replace('_', "-");
         stream.raw("<");
-        stream.escaped(name);
+        stream.escaped(&element_name);
         if let Some(id) = id {
             stream.raw(" id=\"");
             stream.expr(id);
@@ -154,7 +155,7 @@ impl Generate for Element {
         children.generate(stream);
         if *close {
             stream.raw("</");
-            stream.escaped(name);
+            stream.escaped(&element_name);
             stream.raw(">");
         }
     }
```

That allows this:

```rust
            markdown_toolbar[for="body"] {}
```

to render as desired:

```html
<markdown-toolbar for="body"></markdown-toolbar>
```

It does impose an extra allocation per element, which could be avoided in the common case with a wrapper function that returns a `Cow` if there are no '_' present. If you'd be open to a PR exploring this change I'd be happy to polish it up and submit one.

**Comment by wezm** [comment]
Optimised it a bit more and pushed the changes to my fork: https://github.com/wezm/markup.rs/commit/19cfdf889d8e580bd0133d8a4d3a869f8250d518

**Comment by utkarshkukreti** [comment]
I've added a syntax that also fixes #11. Your example can now be written as:

```
$"markdown-toolbar"[for="textarea_id"] {
    $"md-bold" { "bold" }
    $"md-header" { "header" }
    $"md-italic" { "italic" }
}
```
